app/piping/plumber.py

```diff
@@ -0,0 +1,35 @@
+import re
+
+
+class Plumber(object):
+    def __init__(self, context):
+        self._context = context
+
+    def plumb_item(self, item):
+        interesting_properties = self.get_properties_of_interest(item)
+
+        for interesting_property in interesting_properties:
+            if self._needs_plumbing(item, interesting_property):
+                self._plumb(item, interesting_property)
+
+    def get_properties_of_interest(self, item):
+        if hasattr(item, 'templatable_properties'):
```

Comment:
Would it be possible to have templatable_propertes value on all aspects of the schema? We could set it to [] to start with?